### DISCUSSION TITLE
minimum version of macos to compile ipa for ios15

### DISCUSSION THREAD

**Discussion by jebarbieri** [airsdk/Adobe-Runtime-Support#1891] [Q&A]
what is the minimum version of macos to compile ipa for ios15?

**Comment by ajwfrost** [comment]
I'd hope that the limitation isn't macOS, but the AIR SDK which needs to have the appropriate settings/configuration and generate appropriate binaries. So, if you're using the latest AIR SDK on macOS 10.12, you should still be able to create an IPA file and have it run on iOS 15.

thanks

**Reply by jebarbieri** [reply]
Thanks for the answer. I have a macbook pro macOS high sierra 10.13.6, I'm using air sdk 821. But the version of xcode on macOS is 10.1 with this configuration, can I compile for ios 15 with this configuration? I have this doubt because in Release_Notes_AIR_SDK_33.1.1.821 it says that the xcode is 13.

<img width="1440" alt="Captura de Tela 2022-05-06 às 10 28 25" src="https://user-images.githubusercontent.com/38330359/167149956-b5541253-0451-46fa-877d-e8910a1995c0.png">
<img width="1440" alt="Captura de Tela 2022-05-06 às 11 10 32" src="https://user-images.githubusercontent.com/38330359/167150027-618dfdca-0641-4ce6-9f55-12d5e3343a63.png">
<img width="1440" alt="Captura de Tela 2022-05-06 às 11 11 12" src="https://user-images.githubusercontent.com/38330359/167150125-1eb6f97b-96f8-46e3-b562-f89fd9069178.png">

**Reply by ajwfrost** [reply]
Hi

The "Build Environment" is the environment that we use ourselves, nothing to do with what you need to have locally... when you create an IPA file using AIR, you don't actually use Xcode at all (you can do it on Windows too), but you would just be linking together libraries that had been built with Xcode 13. The iOS SDK 15.0 is used to generate our 'stub' files which are also then used to link in to the final binary which is then packaged up into an IPA file with metadata that identifies it as being built by an appropriate Xcode/macOS platform.

So it's actually irrelevant what version of macOS you're running on, what version of Xcode you have there, or even whether it's a mac or a Windows machine. The appropriate details are just embedded into the binaries that we provide within the AIR SDK and into the files that the AIR SDK then generates...

Hope that helps!

thanks

  Andrew

**Reply by jebarbieri** [reply]
Thank you very much. So xcode Version 10.1 installed on macOS doesn't interfere with compilation at all? In this case I must use the Apple Distribution certificate for use with Xcode 11 or later or iOS Distribution?

<img width="1440" alt="Captura de Tela 2022-05-10 às 10 53 10" src="https://user-images.githubusercontent.com/38330359/167645083-f91922de-5a45-4d97-afba-d1b41a9c1a24.png">

**Reply by ajwfrost** [reply]
Yes, it doesn't matter what version of Xcode is on the machine - but also, yes, the Apple Development/Distribution certificates can be used with the AIR SDK, per your screenshot
thanks

**Reply by jebarbieri** [reply]
Thank you so much. I installed on ios 15 device.
